nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -103,9 +113,26 @@ class SimpleActionServer
       action_name_,
       std::bind(&SimpleActionServer::handle_goal, this, _1, _2),
       std::bind(&SimpleActionServer::handle_cancel, this, _1),
-      std::bind(&SimpleActionServer::handle_accepted, this, _1));
+      std::bind(&SimpleActionServer::handle_accepted, this, _1),
+      rcl_action_server_get_default_options(),
+      callback_group_);
+    if (spin_thread_) {
+      auto executor_ = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
```

Comment:
Remove `auto`, that will instantiate a local executor_ copy, not initialize the class variable.

Reply:
this is my mistake. if i use `NodeThread` , we don't need global variable  `executor_ ` , just store shared_ptr of executor in `NodeThread` .